CONTRIBUTING_GO.md

```diff
@@ -161,6 +164,47 @@ reasons why this is _the right thing to do_:
 License: MIT
 Signed-off-by: Juan Benet <juan@ipfs.io>
 ```
+
+## Contributor Principles
+
+These principles are a living document and should be updated as-needed to better serve the go-ipfs community.
+
+In order of precedence:
+
+1. **Build the distributed peer-to-peer web.** This is our objective and aim - everything we do is in service of this goal and we only compromise our other principles if needed to achieve this.
+2. **Don't break the IPFS ecosystem.** This isn't "don't break protocols"; we do occasionally break protocols and APIs (with care) when we know that users won't notice it. However, users trust us to not break their workflows/projects. If we do, we'll lose that trust and those users.
+3. **Take ownership.** If you're working on something, own it and see it through. Given that (a) we don't have managers and (b) we don't work synchronously, coordination can be difficult. To make up for this, if you want something to happen, you have to make it happen (that may mean calling a meeting, bugging people, fixing unrelated issues, designing alternative solutions, etc.).
+4. **Ask for forgiveness, not permission.** To enable P.3, if you're having trouble getting people to move on something, err on the side of getting things done (bounded by P.1 and P.2). Note: This doesn't apply when you *know* you'll need to ask for forgiveness... use your best judgment.
+
+## Code Change Policy
+
+The go code change policy must balance "not breaking things" with "getting things done" and should be read in that spirit. Use your best judgment when making changes and PRs; these rules are not set in stone.
+
+**How can the codebase be changed?**
+
+Except in special cases (and mass gx-workspace updates), all changes go through PRs.
```

Comment:
It's probably not good to allow people in general to Gx-update directly in master (or do any depedency change directly to master). Right now it is very easy to screw dependencies if you're not an expert. I suggest we ask that all changes go through PRs (including Gx/deps updates), and that in the general case no PR should be merged with failing CI (or at least without verifying and explaining why CI is red). 

(I understand some of us have the experience to bend this rule in some occasions because we can identify risks better and we know how to find and fix our screw ups, but I would not generalize it).

Reply:
You're probably right. And really, it's not *that* hard to do a PR. I just really wish there were a nice way to query PR status from the command line.

Reply:
CI status, really (although I do run the tests locally so it's not *that* much of an issue).

Reply:
`hub ci-status -v`?

Reply:
Ok, I *really* need to read that man page.